### DISCUSSION TITLE
Infinite scroll paging

### DISCUSSION THREAD

**Discussion by jonathan-pyt** [supabase/supabase#2141] [Questions]
Hello !

I'm new to supabase, currently developing a React native app and want to add infinite scrolling? I think the cursor pagination will be good for this https://www.postgresql.org/docs/9.2/plpgsql-cursors.html
but I can't find any example to implement this.

Can you help?

Greetings,

**Comment by silentworks** [comment]
I think for this it would work in similar way to pagination. Take a look at the `.range` method, you can see how to get it working by looking at this other discussion I answered https://github.com/supabase/supabase/discussions/1223#discussioncomment-641447

**Reply by jonathan-pyt** [reply]
Hi ! Thank you for your reply

> Cursors have the desirable property of pagination consistency on arbitrary queries, showing results as they exist at the time the transaction was started. The isolation level (link is external) of the transaction guarantees that our paginated view of the results won’t change.

only `.range` , won't work well because if someone else add or delete elements at the same time we could have duplication in this list for example ... 
But i think `.range` could be a solution if i made this scrollable list realtime with `on().subscribe()` i guess 🤔

**Reply by silentworks** [reply]
I cannot see how anything would change if someone was to add or delete a element, the query isn't realtime so this shouldn't make a change to your scrolling.
